go/store/nbs/table_set.go

```diff
@@ -132,70 +133,82 @@ func (ts tableSet) get(ctx context.Context, h addr, stats *Stats) ([]byte, error
 	return f(ts.upstream)
 }
 
-func (ts tableSet) getMany(ctx context.Context, reqs []getRecord, foundChunks chan<- *chunks.Chunk, wg *sync.WaitGroup, ae *atomicerr.AtomicError, stats *Stats) bool {
-	f := func(css chunkSources) bool {
+func (ts tableSet) getMany(ctx context.Context, eg *errgroup.Group, reqs []getRecord, found func(*chunks.Chunk), stats *Stats) (bool, error) {
+	f := func(css chunkSources) (bool, error) {
 		for _, haver := range css {
-			if ae.IsSet() {
-				return false
-			}
-
 			if rp, ok := haver.(chunkReadPlanner); ok {
 				offsets, remaining := rp.findOffsets(reqs)
-
-				rp.getManyAtOffsets(ctx, reqs, offsets, foundChunks, wg, ae, stats)
-
+				err := rp.getManyAtOffsets(ctx, eg, offsets, found, stats)
+				if err != nil {
+					return true, err
+				}
 				if !remaining {
-					return false
+					return false, nil
 				}
-
 				continue
 			}
 
-			remaining := haver.getMany(ctx, reqs, foundChunks, wg, ae, stats)
-
+			remaining, err := haver.getMany(ctx, eg, reqs, found, stats)
+			if err != nil {
+				return true, err
+			}
 			if !remaining {
-				return false
+				return false, nil
 			}
 		}
-
-		return true
+		return true, nil
 	}
 
-	return f(ts.novel) && f(ts.upstream)
+	remaining, err := f(ts.novel)
```

Comment:
A potential downside of this errgroup change is that now we have synchronous and asynchronous errors reported differently on paths like this one. It makes things like this here a little awkward.

Reply:
Could always do something like:

```
if err != nil {
    eg.Go(func() error {
        return err
    }

    return false
}```

if you don't like the mix... But that seems worse.